client/src/controllers/CountController.ts

```diff
@@ -1,7 +1,8 @@
 import { Controller } from '@hotwired/stimulus';
 import { ngettext } from '../utils/gettext';
 
-const DEFAULT_ERROR_SELECTOR = '.error-message,.help-critical';
+const DEFAULT_ERROR_SELECTOR =
+  ':not([hidden]):is(.error-message,.help-critical)';
```

Comment:
This doesn't handle messages that are inside a hidden element (i.e. the ancestor is hidden, not necessarily the error message itself).

I tried handling that as well but it seems the https://github.com/dperini/nwsapi used in testing doesn't like complex selectors. It didn't even like if I put the `:is()` first, i.e. `:is(.error-message,.help-critical):not([hidden])`.

If you think that should be handled as well, and managed to get a working selector, please feel free to update the PR.